trunk/source/WitnessPrinter/src/de/uni_freiburg/informatik/ultimate/witnessprinter/yaml/YamlWitnessWriterV2.java

```diff
@@ -47,24 +50,67 @@
  */
 public class YamlWitnessWriterV2 extends YamlWitnessWriter {
 	private final MetadataProvider mMetadataProvider;
-	private final boolean mWriteFunctionContracts;
+	private final boolean mWriteFunctionContractsAndGhostVariables;
```

Comment:
shouldn't this be two separate settings?

Reply:
Or perhaps rather, does it even make sense to just filter out ghost variables here? The invariants may still refer to them, making the generated witness invalid. If we want to sometimes avoid ghost variables, we may have to do a more complex transformation earlier on (e.g. by quantifying the variables in invariants).

Reply:
The intention here was to generate ghost variables and function contracts only for version 2.1, since the format in 2.0 does not allow ghost variables. But you are right, we can not simply ommit ghost variables in the witness, but keep them in invariants. So I think, we should just crash, if we want to print a witness with ghost in version 2.0.